packages/graph/hash_graph/lib/graph/src/ontology/data_type.rs

```diff
@@ -24,6 +29,251 @@ pub enum DataTypeQueryPath<'q> {
+                })
+                .collect(),
+        }
+    }
+
+    fn convert_path(
+        segments: impl IntoIterator<Item = &'static str>,
+    ) -> DataTypeQueryPath<'static> {
+        DataTypeQueryPath::try_from(create_path(segments)).expect("Could not convert path")
+    }
+
+    fn deserialize<'q>(segments: impl IntoIterator<Item = &'q str>) -> DataTypeQueryPath<'q> {
+        DataTypeQueryPath::deserialize(de::value::SeqDeserializer::<_, de::value::Error>::new(
+            segments.into_iter(),
+        ))
+        .expect("Could not deserialize path")
+    }
+
+    #[test]
+    fn deserialization() {
+        assert_eq!(deserialize(["baseUri"]), DataTypeQueryPath::BaseUri);
+        assert_eq!(deserialize(["version"]), DataTypeQueryPath::Version);
+        assert_eq!(
+            deserialize(["versionedUri"]),
+            DataTypeQueryPath::VersionedUri
+        );
+        assert_eq!(deserialize(["ownedById"]), DataTypeQueryPath::OwnedById);
+        assert_eq!(deserialize(["type"]), DataTypeQueryPath::Type);
+        assert_eq!(deserialize(["title"]), DataTypeQueryPath::Title);
+        assert_eq!(deserialize(["description"]), DataTypeQueryPath::Description);
+        assert_eq!(
+            deserialize(["custom"]),
+            DataTypeQueryPath::Custom(Cow::Borrowed("custom"))
+        );
+
+        assert_eq!(
+            DataTypeQueryPath::deserialize(de::value::SeqDeserializer::<_, de::value::Error>::new(
+                ["baseUri", "test"].into_iter()
+            ))
+            .expect_err("Could convert data type query path with multiple tokens")
+            .to_string(),
+            "invalid length 2, expected 1 element in sequence"
+        );
+    }
+
+    #[test]
+    fn path_conversion() {
+        assert_eq!(convert_path(["baseUri"]), DataTypeQueryPath::BaseUri);
+        assert_eq!(convert_path(["version"]), DataTypeQueryPath::Version);
+        assert_eq!(
+            convert_path(["versionedUri"]),
+            DataTypeQueryPath::VersionedUri
+        );
+        assert_eq!(convert_path(["ownedById"]), DataTypeQueryPath::OwnedById);
+        assert_eq!(convert_path(["type"]), DataTypeQueryPath::Type);
+        assert_eq!(convert_path(["title"]), DataTypeQueryPath::Title);
+        assert_eq!(
+            convert_path(["description"]),
+            DataTypeQueryPath::Description
+        );
+        assert_eq!(
+            convert_path(["custom"]),
+            DataTypeQueryPath::Custom(Cow::Borrowed("custom"))
+        );
+
+        assert_eq!(
+            DataTypeQueryPath::try_from(create_path(["baseUri", "invalid"]))
+                .expect_err("Could convert data type query path with multiple tokens")
+                .downcast_ref::<de::value::Error>()
+                .expect("deserialization error not found in report")
+                .to_string(),
+            "invalid length 2, expected 1 element in sequence"
+        );
+    }
+}
+
 impl QueryRecord for DataType {
```

Comment:
Another nit:
This impl gets a little lost below the tests, perhaps we should move it up :)

Reply:
Whoops, fixed 🙂